curvefs/src/mds/mds_service.cpp

```diff
@@ -275,8 +275,8 @@ void MdsServiceImpl::AllocateS3Chunk(
     brpc::ClosureGuard guard(done);
     VLOG(9) << "start to allocate chunkId.";
 
-    uint64_t chunkId = 0;
-    int stat = chunkIdAllocator_->GenChunkId(&chunkId);
+    uint64_t  chunkId = 0;
+    int stat = chunkIdAllocator_->GenChunkId(request->chunkidnum(), &chunkId);
```

Comment:
should you test whether the request has chunkidnum, otherwise, it will return 0, or just give the optional field a default value in the message definition